api/system-roles/update

```diff
@@ -25,30 +25,46 @@ require "/usr/libexec/nethserver/api/lib/auth_functions.pl";
 
 use warnings;
 use strict;
-use JSON;
+use esmith::ConfigDB;
+my $db = esmith::ConfigDB->open();
 
-my $jsonFile = '/etc/nethserver/cockpit/authorization/roles.json';
 my $input = readInput();
 my $role = $input->{'role'} || '';
 my $groupAdmins = get_adminGroup();
 
-my $data = read_json ($jsonFile); 
+my @system = @{$input->{'system'}}; 
+my @applications = @{$input->{'applications'}};
+my @permissions = ();
+push @permissions,(@system,@applications);
+my $values = '';
 
 if ($role ne '') {
 
     if ($role ne  $groupAdmins) {
-        $data->{$role}->{'system'} = $input->{'system'};
-        $data->{$role}->{'applications'} = $input->{'applications'};
-        write_json ($jsonFile,$data);
+        my @Keys = split(',', $db->get_prop('cockpit.socket','delegation')|| '');
+        foreach (@Keys) {
+            my @Props = split(':',$_);
+            if ($Props[0] eq $role) { # bingo we find the values to update
+                my $group = shift(@Props);
+                $values .= $group.':'.join(':',@permissions).',';
+            }else { # this value is not the one to update, push it to values
+               $values .= $_.',';
+            }
+        }
+        # we did not find the values to update above, 
+        # search and if it doesn't exist, write it to db
+        if ($values !~ /,$role:|^$role:/) {
+             $values .= $role.':'. join(':',@permissions).','; 
+        }
+        chop ($values);
+        $db->set_prop('cockpit.socket','delegation', $values); 
+        my $out = `/sbin/e-smith/expand-template /etc/sudoers.d/50_servermanager_perms`;
```

Comment:
An API script usually does not expand a template in this way, it raises an esmith event instead. You can use the `esmith::event` library for that.